### PR TITLE
[server] infer extensions and add a comment

### PR DESCRIPTION
**Pull request by svenefftinge**
This commit adds vscode extensions to the inferred `.gitpod.yml` and adds a comment to the top, that links to the documentation.

follow up to: https://github.com/gitpod-io/gitpod/pull/7383

## How to test
Start workspaces on repositories without `.gitpod.yml` and check the generated `.gitpod.yml`.

## Release Notes
```release-note
Also propose vscode extensions for yet unconfigured repositories.
```

### CODE REVIEW

**Comment by roboquat** [comment]
@svenefftinge: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **alextugarev** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @alextugarev` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jankeromnes — COMMENTED** [review]
Many thanks!

Code looks good to me at first glance. 👍

/approve

I wanted to test this, but couldn't start a workspace:

<table><tr><td>
Screenshot 2022-01-03 at 12 00 33
</td><td>
Screenshot 2022-01-03 at 12 00 42
</td><td>
Screenshot 2022-01-03 at 12 00 57
</td></tr></table>

Maybe an unstable deployment?

Re-triggering just in case:

/werft run

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7392#pullrequestreview-842617126" title="Approved">jankeromnes</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/config/config-inferrer.ts
Comment (akosyakov) [inline components/server/src/config/config-inferrer.ts:42]: We already always install it in VS Code.

**Comment by akosyakov** [comment]
@svenefftinge You should look at product.json of VS Code in your machine for inspiration of hints based on repo content.

**Comment by svenefftinge** [comment]
> @svenefftinge You should look at product.json of VS Code in your machine for inspiration of hints based on repo content.

Thanks, it's not the main intention to of this PR to have a complete list, but rather make a few non-controversial suggestions. Should definitely be further improved over time.

File: components/server/src/config/config-inferrer.ts
Comment (svenefftinge) [reply components/server/src/config/config-inferrer.ts]: Removed it 🙏

File: components/server/src/config/configuration-service.ts
Comment (svenefftinge) [reply components/server/src/config/configuration-service.ts:50]: I'd like to keep it in parenthesis if you can live with that. there are no par5sing issues in vscode at least.

File: components/server/src/config/configuration-service.ts
Comment (svenefftinge) [reply components/server/src/config/configuration-service.ts]: Thanks & applied

File: components/server/src/config/configuration-service.ts
Comment (jankeromnes) [inline components/server/src/config/configuration-service.ts:51]: I'm not sure about the phrasing "to share the goodness with others", but I don't have a better proposal right now.

@gtsiolis Would you have any suggestions to improve this comment? (It's added to every auto-generated `.gitpod.yml` when you open a Gitpod workspace for any repository that doesn't already have one.)

Note: Let's not block this PR, but work on a better comment in a follow-up PR.

File: components/server/src/config/configuration-service.ts
Comment (jankeromnes) [inline components/server/src/config/configuration-service.ts:54]: No need to add a second empty line at the bottom of the file (can also be fixed in a follow-up):

```suggestion
${configString}`;
```

**Review by jankeromnes — APPROVED** [review]
Looks good to me and works like a charm! 🧙

I see a few more potential small improvements, but these can definitely be done as follow-ups, so unblocking this outstanding PR. 🚀

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 38c422ae7e3d40a90361bf013b266762d55d97c9</details>

**Pull request merged**
